ingest/expression_files/expression_files.py

```diff
@@ -85,15 +94,17 @@ def set_data_array_gene_cell_names(
             linear_data_id,
             self.study_id,
             self.study_file_id,
-        ).get_data_array()
+        ).get_data_array():
+            yield model
```

Comment:
`set_data_array_cells` and `set_data_array_gene_cell_names` looks mostly redundant.  

Using `self.cluster_name` if `name` is `None` or some such seems like a possible way to consolidate these two methods.

If consolidation is unworkable, then using a consistent order for the two methods' seems worthwhile.  (One has `values` then `linear_data_id` in its signature, the other has `linear_data_id` then `values`; using a single consistent ordering seems more maintainable.)

Reply:
The structure of the objects yielded by both functions is the same - there are simply some values that get set by default that are different between the two.  Specifically the `name` of the array, and the `array_type` of either `expression` or `cells`.  I would support merging them together as well.